### PR TITLE
PART 2 - 957 feature request production method in steel cross section

### LINKED ISSUE #957: [✨ Feature request]: production_method in steel cross section
### Description

List production_method in the steel cross section such that we know if its rolled or welded.

Needed for 1993-1-1 2005 eq 6.2.6 (3)

### Acceptance Criteria

- code
- tests

### Dependencies

_No response_

### Confirmation

- [x] I have checked that this feature does not already exist in the issue tracker.
- [x] I have read and understood the contribution guidelines.

### PR DESCRIPTION
**Pull request by GerjanDorgelo**
## Description

feature request production method in steel cross section

Fixes #957 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

* **New Features**
  * Steel cross sections now support designation of section type, allowing users to specify whether sections are rolled or welded (default: rolled).

* **Tests**
  * Added test coverage to verify section type functionality.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by SZeltaat — COMMENTED** [review]
Hi Gerjan,
I think this is the right place for this attribute, but we should be more carefull with the default. Please see my comment 😉

File: blueprints/structural_sections/steel/steel_cross_section.py [hunk 1]
```diff
@@ -21,12 +22,16 @@ class SteelCrossSection:
...
         The material type of the steel.
+    fabrication_method : Literal["rolled", "welded"], optional
+        The fabrication method of the steel cross-section, either "rolled" or "welded". Default is "rolled".
     """
 
     profile: Profile
     """The profile. This can be a predefined profile or a generic profile."""
     material: SteelMaterial
     """The material type of the steel."""
+    fabrication_method: Literal["rolled", "welded"] = "rolled"
```
[the remark this answers is not in the document]
Comment (GerjanDorgelo) [reply blueprints/structural_sections/steel/steel_cross_section.py]: I think later might be now (for the H-profiles). I need it for my I beam shear branch. Would you be okay with a None as default, but the HEA/B/M and IPE profiles i set them as hot-rolled? The cannot be cold-formed as its not possible to make an H form out of a flat sheet of steel.

And while im on it. RHS and SHS are hot-rolled. RHSCF and SHSCF are cold formed.

**Review by SZeltaat — CHANGES_REQUESTED** [review]
I think this can be implemented in a much better way. Now, we look for profile's name in the whole database. We can narrow it down more effectively.

In any case, we should have good test cases to test this functionality.
- For hot rolled
- For cold-formed
- For edge cases where name is not found, etc... (so every if-condition)
- For tricky situations like RHS and RHSCF

If all test cases pass, we can always change the implementation later.



My suggestion for the implementation:

First extract only the first alphabetic part of the name.
Find the corresponding profile class by matching against the cls.__name__
Then extract the profile name (like you already do, alternatively use the pattern alphabetic, followed by multiple numerics optionally seperated by "x")
Then if any(profile_name == standard_profile.name for standard_profile in cls._database.values()), assign hot-rolled / cold-formed based on cls.__name__.

**Review by SZeltaat — APPROVED** [review]
LGTM!
